pipeline-adapters/mlrun-pipelines-kfp-v1-8/src/mlrun_pipelines/client.py

```diff
@@ -664,6 +664,41 @@ def terminate_run(
             )
             raise error
 
+    def get_experiment_with_project_name(
+        self,
+        project_name: str,
+    ) -> kfp_server_api.ApiExperiment:
+        """
+        Retrieve an experiment by project name.
+        This method searches for an experiment whose name matches the project name,
+        allowing for a dash-prefixed match (e.g., "myproject-").
+        :param project_name: The name of the project to search for.
+        :return: An ApiExperiment object representing the found experiment.
+        :raises ValueError: If no experiment is found with the specified project name.
+        """
+
+        filter_json = orjson.dumps(
+            {
+                "predicates": [
+                    {
+                        "key": "name",
+                        "op": mlrun_pipelines.models.FilterOperations.IS_SUBSTRING,
+                        "string_value": project_name,
+                    }
+                ]
+            }
+        ).decode()
+
+        experiments = (
+            self._experiment_api.list_experiment(filter=filter_json).experiments or []
+        )
+        for experiment in experiments:
+            normalized_name = experiment.name.rsplit("-", 1)[0]
+            if project_name in (normalized_name, experiment.name):
+                return experiment
```

Comment:
I think this is problematic and may cause us to lose results. The experiment name is:
`experiment = name or workflow_spec.name`
So it can either be name or the workflow name, and the name comes from:
`name = f"{self.metadata.name}-{name}" if name else self.metadata.name`
in project.run(). So depending on how users run their pipelines, a single project can have more than one experiment name.
this function only finds ONE experiment per project, meaning users will lose visibility into runs from other naming patterns.